relay-server/src/services/processor.rs

```diff
@@ -1895,17 +1856,37 @@ impl EnvelopeProcessorService {
                 .parametrize_dsc_transaction(&sampling_state.config.tx_name_rules);
         }
 
+        // Set the event retention. Effectively, this value will only be available in processing
+        // mode when the full project config is queried from the upstream.
+        if let Some(retention) = project_info.config.event_retention {
```

Comment:
Moved this here from the `prepare_state` since it has nothing to do with state preparation, it's just part of the initialization.